src/metadata.rs

```diff
@@ -112,6 +120,24 @@ pub fn get_ids(root_path: &str) -> io::Result<Vec<String>> {
         .collect::<Vec<String>>())
 }
 
+pub fn get_valid_id(id: &str) -> io::Result<String> {
+    let s = String::from(id.trim());
+    if is_packet_str(&s) {
+        Ok(s)
+    } else {
+        Err(io::Error::new(io::ErrorKind::InvalidInput,
+                           format!("Invalid packet id '{}'", id)))
+    }
+}
+
+pub fn get_missing_ids(root_path: &str, wanted: &str, unpacked: Option<bool>) -> io::Result<Vec<String>> {
```

Comment:
I think that for the metadata functions, this should be taking something like `&[&str]` or `&[String]` with the API part taking responsibility for the split